libraries/stdlib/js/src/kotlin/string.kt

```diff
@@ -46,3 +46,11 @@ public inline val CharSequence.size: Int get() = length
 
 @kotlin.internal.InlineOnly
 internal inline fun String.nativeReplace(pattern: RegExp, replacement: String): String = asDynamic().replace(pattern, replacement)
+
+/**
+ * Converts the characters from a portion of the specified array to a string.
+ */
+@kotlin.internal.InlineOnly
+public actual inline fun String(chars: CharArray, offset: Int, length: Int): String {
+    return chars.joinToString(separator = "").substring(offset, offset + length)
```

Comment:
Not a very efficient way to build a string from a `CharArray` especially if the `length` is way more smaller than the length of the array.

Reply:
How about this?
```kotlin
return chars.sliceArray(offset until offset + length).joinToString(separator = "")
```

Reply:
I mean something like that: https://developers.google.com/web/updates/2012/06/How-to-convert-ArrayBuffer-to-and-from-String